src/main/java/com/genesys/statistics/Notifications.java

```diff
@@ -0,0 +1,129 @@
+package com.genesys.statistics;
+
+import java.net.HttpCookie;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.cometd.bayeux.Message;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Notifications {
+    private static final Logger logger = LoggerFactory.getLogger(Notifications.class);
+
+    boolean isInitialized() {
+        return client != null;
+    }
+
+    public static interface NotificationListener {
+        void onNotification(String channel, Map<String,Object> data);
+    }
+
+    private BayeuxClient client;
+    private HttpClient httpClient;
+    private HttpCookie session;
+    
+    private final Map<String, List<NotificationListener>> listeners = new HashMap<>();
```

Comment:
use Map<String, Collection< NotificationListener >> and create [ConcurrentHashMap]  (https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html)